src/Tools/dotnet-monitor/CollectionRules/CollectionRuleService.cs

```diff
@@ -155,6 +159,11 @@ private async Task MonitorRuleChangesAsync(CancellationToken token)
 
                 _logger.CollectionRuleConfigurationChanged();
 
+                if (DiagnosticPortOptionsExtensions.GetConnectionMode(_portOptions) != DiagnosticPortConnectionMode.Listen)
```

Comment:
Couple of things:
- Move this to a private method.
- Call it from the ctor and from this location.
- Update the code to check if there are any collection rules before logging the warning. If the user doesn't have collection rules, no point in logging the warning.